src/hotspot/share/opto/ifnode.cpp

```diff
@@ -2181,6 +2211,8 @@ void ParsePredicateNode::dump_spec(outputStream* st) const {
     default:
       fatal("unknown kind");
   }
+  if (_useless) {
+    st->print("#useless ");
+  }
```

Comment:
Squeezed this in here for Parse Predicates. It's probably not worth to create a separate RFE for that.